src/delegation/store/memory.rs

```diff
@@ -79,7 +80,7 @@ pub struct MemoryStore<
     V: varsig::Header<C> = varsig::header::Preset,
     C: Codec + TryFrom<u64> + Into<u64> = varsig::encoding::Preset,
 > {
-    inner: Arc<RwLock<MemoryStoreInner<DID, V, C>>>,
+    inner: Arc<Mutex<MemoryStoreInner<DID, V, C>>>,
```

Comment:
To account for starvation I guess?

Reply:
Yeah, due to [this conversation](https://github.com/fission-codes/rs-ucan/pull/9#discussion_r1534768116). I think RwLocks being able to cause deadlocks is... not great. Defaulting to Mutex seems like the better option :+1: